contracts/money-market/PriceOracle/PriceOracleProxyUSD.sol

```diff
@@ -0,0 +1,193 @@
+        require(price > 0, "invalid price");
+
+        // Extend the decimals to 1e18.
+        return mul_(uint256(price), 10**(18 - uint256(aggregator.decimals())));
+    }
+
+    /**
+     * @notice Get price from v1 price oracle
+     * @param bTokenAddress The BToken address
+     * @return The price
+     */
+    function getPriceFromV1(address bTokenAddress)
+        internal
+        view
+        returns (uint256)
+    {
+        address underlying = BErc20(bTokenAddress).underlying();
+        return v1PriceOracle.assetPrices(underlying);
+    }
+
+    /*** Admin or guardian functions ***/
+
+    event AggregatorUpdated(
+        address bTokenAddress,
+        address source,
+        AggregatorBase base
+    );
+    event SetGuardian(address guardian);
+    event SetAdmin(address admin);
+
+    /**
+     * @notice Set guardian for price oracle proxy
+     * @param _guardian The new guardian
+     */
+    function _setGuardian(address _guardian) external {
+        require(msg.sender == admin, "only the admin may set new guardian");
+        guardian = _guardian;
+        emit SetGuardian(guardian);
+    }
+
+    /**
+     * @notice Set admin for price oracle proxy
+     * @param _admin The new admin
+     */
+    function _setAdmin(address _admin) external {
+        require(msg.sender == admin, "only the admin may set new admin");
+        admin = _admin;
+        emit SetAdmin(admin);
+    }
+
+    /**
+     * @notice Set ChainLink aggregators for multiple bTokens
+     * @param bTokenAddresses The list of bTokens
+     * @param sources The list of ChainLink aggregator sources
+     * @param bases The list of ChainLink aggregator bases
+     */
+    function _setAggregators(
+        address[] calldata bTokenAddresses,
+        address[] calldata sources,
+        AggregatorBase[] calldata bases
+    ) external {
+        require(
+            msg.sender == admin || msg.sender == guardian,
+            "only the admin or guardian may set the aggregators"
+        );
+        require(
+            bTokenAddresses.length == sources.length &&
+                bTokenAddresses.length == bases.length,
+            "mismatched data"
+        );
+        for (uint256 i = 0; i < bTokenAddresses.length; i++) {
+            AggregatorV3Interface source = AggregatorV3Interface(sources[i]);
+            
+            if (sources[i] != address(0)) {
+                require(
+                    msg.sender == admin,
+                    "guardian may only clear the aggregator"
+                );
+
+                (, int256 answer,,,) = source.latestRoundData();
+                require(
+                    answer > 0,
+                    "invalid pricing aggregator"
+                );
+            }
+
+            aggregators[bTokenAddresses[i]] = AggregatorInfo({
+                source: source,
+                base: bases[i]
+            });
+            emit AggregatorUpdated(bTokenAddresses[i], sources[i], bases[i]);
+        }
+    }
```

Comment:
We should explicitly support only two `bases`: USD and ETH. [The Iron Bank version](https://github.com/ibdotxyz/compound-protocol/blob/6e1d3d350147d53834ca3cd6e9a068592f6e2654/contracts/PriceOracle/PriceOracleProxy.sol#L308) filters on this basis, and it is needed here as well because we implicitly assume in `getUnderlyingPrice()` that all non-ETH bases are USD.